gokart/parameter.py

```diff
@@ -36,6 +47,12 @@ def serialize(self, x):
         values = dict(type=x.get_task_family(), params=params)
         return luigi.DictParameter().serialize(values)
 
+    def _warn_on_wrong_param_type(self, param_name, param_value):
+        if self.__class__ != TaskInstanceParameter:
```

Comment:
I think this return is for Paramteres who has inherited TaskInstanceParameter.

I think expected_type check is also important for those parameters, so removing this maybe considerable.

Reply:
Thank you for the comment! I've removed this class check